### PR TITLE
Add trigger service step

### PR DESCRIPTION
**Pull request by krauselukas**
This PR implement a step that enables users to trigger services through their workflows.

*TODO*

- [x] Write and adapt specs
- [x] Tackle Rubocop offenses 
- [x] Testing with a real setup

### CODE REVIEW

File: src/api/app/models/token/workflow.rb [hunk 1]
```diff
@@ -31,11 +31,6 @@ def call(options)
     raise Token::Errors::SCMTokenInvalid, e.message
   end
 
-  # Only used by rebuild steps
-  def package_find_options
-    { use_source: false, follow_project_links: true, follow_multibuild: true }
```
Comment (hennevogel) [inline src/api/app/models/token/workflow.rb:36]: You noticed that they differ per token type right?

Comment (krauselukas) [reply src/api/app/models/token/workflow.rb:36]: Yes, thats why I moved it from the workflow token model to the individual steps, in order to pass the correct options for the Triggerable concern

File: src/api/app/controllers/concerns/triggerable.rb [hunk 2]
```diff
@@ -27,6 +28,7 @@ def set_package
     # TODO: This should not happen right? But who knows...
     raise ActiveRecord::RecordNotFound unless @package
   end
+  # rubocop:enable Naming/AccessorMethodName
```
Comment (dmarcoux) [inline src/api/app/controllers/concerns/triggerable.rb:31]: This comment could be right under `def set_package(package_find_options:)`.

Comment (krauselukas) [reply src/api/app/controllers/concerns/triggerable.rb]: Done

**Comment by saraycp** [comment]
I've done a few manual tests locally and worked fine. I'd like to test more, but for now, I've detected a few typos in commit descriptions:

First commit:
- replace "_Since we going_" with "_Since we are going_"
- replace  "_we going to have more then_" with "_we are going to have more than_"

Second commit:
- replace "_inorder_" with "_in order_"

I also wonder if it makes sense to rescue from possible exceptions raised by the [backend call](https://github.com/openSUSE/open-build-service/pull/12362/files#diff-0640a1ff102c6e55ae84ad3a8b8f55ebd971c99a11a338c90a785263f24a6ebbR18).
In my case I got one for not having the service correctly set up:

```
Backend::NotFoundError: <status code="404">
  <summary>no source service defined!</summary>
  <details>404 no source service defined!</details>
</status>
```
 I know we are not doing so for the [RebuildPackage step](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/workflow/step/rebuild_package.rb#L36), but we rescue from other exceptions in steps like [BranchPackageStep](https://github.com/openSUSE/open-build-service/blob/96148b1a43e9c8d854d37695f04deda03cb04919/src/api/app/models/workflow/step/branch_package_step.rb#L51).

File: src/api/app/controllers/concerns/triggerable.rb [hunk 3]
```diff
@@ -11,13 +11,16 @@ def set_project
     raise Project::Errors::UnknownObjectError, "Sorry, triggering tokens for remote project \"#{@project_name}\" is not possible." unless @project.is_a?(Project)
   end
 
-  def set_package
+  # rubocop:disable Naming/AccessorMethodName
+  def set_package(package_find_options: {})
+    # rubocop:enable Naming/AccessorMethodName
```
Comment (danidoni) [inline src/api/app/controllers/concerns/triggerable.rb:16]:
```suggestion
  def set_package(package_find_options: {}) # rubocop:disable Naming/AccessorMethodName
```

I think this works, too...

Comment (krauselukas) [reply src/api/app/controllers/concerns/triggerable.rb]: done @danidoni

**Review by danidoni — APPROVED** [review]

**Review by saraycp — APPROVED** [review]

**Pull request merged**